qt/widgets/mplcpp/test/MplCppTestInitialization.h

```diff
@@ -62,7 +62,10 @@ class QApplicationHolder : CxxTest::GlobalFixture {
     return true;
   }
 
-private:
+  int m_argc = 1;
+  GNU_DIAG_OFF("pedantic")
+  char *m_argv[1] = {"MplCppTest"};
```

Comment:
Well, we would be constructing something to then pass in a different type? This is some code from @martyngigg that I just plugged in, in my opinion however I find that it makes more sense to define the type of the object you are passing instead of a less efficient and potentially less clear solution.

Reply:
The Qt documentation says they have to remain valid for the lifetime of `QApplication`.

Reply:
And if they were local then they would go out of scope as soon as the method that creates the `QApplication` object.